sprite.go

```diff
@@ -1759,3 +1766,105 @@ func (pself *SpriteImpl) DeltaTime() float64 {
 func (pself *SpriteImpl) TimeSinceLevelLoad() float64 {
 	return time.TimeSinceLevelLoad()
 }
+
+// ------------------------ sound ----------------------------------------
+
+// Play func:
+//
+//	Play(sound)
+//	Play(video) -- maybe
+//	Play(media, wait) -- sync
+//	Play(media, opts)
+func (p *SpriteImpl) Play__0(media Sound) {
+	if media == nil {
+		panic("play media is nil")
+	}
+	p.Play__2(media, &PlayOptions{})
+}
+
+func (p *SpriteImpl) Play__1(media Sound, wait bool) {
+	p.Play__2(media, &PlayOptions{Wait: wait})
+}
+
+func (p *SpriteImpl) Play__2(media Sound, action *PlayOptions) {
+	if debugInstr {
+		log.Println("Play", media.Path)
+	}
+
+	p.checkAudioId()
+	err := p.g.play(p.audioId, media, action)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (p *SpriteImpl) Play__3(media SoundName) {
+	p.Play__5(media, &PlayOptions{})
+}
+
+func (p *SpriteImpl) Play__4(media SoundName, wait bool) {
+	p.Play__5(media, &PlayOptions{Wait: wait})
+}
+
+func (p *SpriteImpl) Play__5(media SoundName, action *PlayOptions) {
+	m, err := p.g.loadSound(media)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	p.Play__2(m, action)
+}
+
+func (p *SpriteImpl) Volume() float64 {
+	return p.g.sounds.getVolume(p.audioId) * 100
+}
+
+func (p *SpriteImpl) SetVolume(volume float64) {
+	p.checkAudioId()
+	val := volume / 100
+	if val <= 0 {
+		val = 0.01
+	}
+	p.g.sounds.setVolume(p.audioId, val)
+}
+
+func (p *SpriteImpl) ChangeVolume(delta float64) {
+	p.checkAudioId()
+	p.SetVolume(p.Volume() + delta)
+}
+
+func (p *SpriteImpl) SoundPan() float64 {
+	return p.g.sounds.getPan(p.audioId) * 100
+}
+
+func (p *SpriteImpl) SetSoundPan(value float64) {
```

Comment:
If we follow the API style defined in spx v1:

https://github.com/goplus/spx/blob/e9e1f63624990d816ffe310a438e19f6aef16ec1/game.go#L1308-L1336

It should be `SetSoundEffect(kind SoundEffectKind, val float64)` here. I wonder why we are not following that.

Reply:
Also related to goplus/spx#531.